From: Baolin Wang <baolin.wang@linux.alibaba.com>
Subject: mm-shmem-split-large-entry-if-the-swapin-folio-is-not-large-fix-2
Date: Tue, 27 Aug 2024 14:46:40 +0800

Now we only split large folio to order 0, so drop the 'new_order'
parameter.

Link: https://lkml.kernel.org/r/39c71ccf-669b-4d9f-923c-f6b9c4ceb8df@linux.alibaba.com
Signed-off-by: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: Barry Song <baohua@kernel.org>
Cc: Chris Li <chrisl@kernel.org>
Cc: Daniel Gomez <da.gomez@samsung.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: "Huang, Ying" <ying.huang@intel.com>
Cc: Hugh Dickins <hughd@google.com>
Cc: Kefeng Wang <wangkefeng.wang@huawei.com>
Cc: Lance Yang <ioworker0@gmail.com>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Pankaj Raghav <p.raghav@samsung.com>
Cc: Ryan Roberts <ryan.roberts@arm.com>
Cc: Yang Shi <shy828301@gmail.com>
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/shmem.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/mm/shmem.c~mm-shmem-split-large-entry-if-the-swapin-folio-is-not-large-fix-2
+++ a/mm/shmem.c
@@ -1996,10 +1996,10 @@ static void shmem_set_folio_swapin_error
 }
 
 static int shmem_split_large_entry(struct inode *inode, pgoff_t index,
-				   swp_entry_t swap, int new_order, gfp_t gfp)
+				   swp_entry_t swap, gfp_t gfp)
 {
 	struct address_space *mapping = inode->i_mapping;
-	XA_STATE_ORDER(xas, &mapping->i_pages, index, new_order);
+	XA_STATE_ORDER(xas, &mapping->i_pages, index, 0);
 	void *alloced_shadow = NULL;
 	int alloced_order = 0, i;
 
@@ -2027,7 +2027,7 @@ static int shmem_split_large_entry(struc
 		}
 
 		/* Try to split large swap entry in pagecache */
-		if (order > 0 && order > new_order) {
+		if (order > 0) {
 			if (!alloced_order) {
 				split_order = order;
 				goto unlock;
@@ -2038,7 +2038,7 @@ static int shmem_split_large_entry(struc
 			 * Re-set the swap entry after splitting, and the swap
 			 * offset of the original large entry must be continuous.
 			 */
-			for (i = 0; i < 1 << order; i += (1 << new_order)) {
+			for (i = 0; i < 1 << order; i++) {
 				pgoff_t aligned_index = round_down(index, 1 << order);
 				swp_entry_t tmp;
 
@@ -2124,7 +2124,7 @@ static int shmem_swapin_folio(struct ino
 		 * should split the large swap entry stored in the pagecache
 		 * if necessary.
 		 */
-		split_order = shmem_split_large_entry(inode, index, swap, 0, gfp);
+		split_order = shmem_split_large_entry(inode, index, swap, gfp);
 		if (split_order < 0) {
 			error = split_order;
 			goto failed;
_
